src/main/java/gameframework/assets/Sound.java

```diff
@@ -0,0 +1,83 @@
+package gameframework.assets;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+
+import javax.sound.sampled.AudioFormat;
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.Clip;
+import javax.sound.sampled.DataLine;
+import javax.sound.sampled.LineUnavailableException;
+import javax.sound.sampled.UnsupportedAudioFileException;
+
+/**
+ * Class allowing to create sounds and play them. It is not intended to play musics.
+ */
+public class Sound {
+
+	/**
+	 * The sound clip instance
+	 */
+	protected final Clip clip;
+
+	/**
+	 * True if the sound must loop, false otherwise
+	 */
+	protected boolean isLooping;
+
+	/**
+	 * Creates and loads the sound
+	 * @param path the path to the sound asset
+	 * @throws IOException
+	 * @throws UnsupportedAudioFileException
+	 * @throws LineUnavailableException
+	 */
+	public Sound(final String path) throws UnsupportedAudioFileException, IOException, LineUnavailableException {
+		final BufferedInputStream bis = new BufferedInputStream(getClass().getResourceAsStream(path));
```

Comment:
The clip variable is final because it should never be reassigned. If you want to change the sound, you must create a new Sound object. Maybe I should have made that clearer in the comments, I'll add it.

About the path variable, it's an eclipse configuration that I have. In general, constructor parameters should not be modified, so making them final assures me that I don't do it by mistake or inattention. I've also been told that it made an optimization for the java compiler, but I've never came accross any reliable source about it.